src/geometry/aabb.rs

```diff
@@ -1,72 +1,163 @@
-//! An axis-aligned cube.
+//! Axis-aligned box and cube.
 
 use crate::math::sat::{self, ConvexPolyhedron, Intersector, Relation};
 use crate::math::PointCulling;
+use crate::proto;
 use arrayvec::ArrayVec;
-use cgmath::{BaseFloat, Point3, Vector3};
-use collision::{Aabb, Contains};
-use num_traits::Bounded;
+use nalgebra::{Isometry3, Point3, RealField, Unit, Vector3};
+use serde::{Deserialize, Serialize};
 
-pub use collision::Aabb3;
+/// An axis-aligned bounding box.
+#[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
```

Comment:
Hmm, all other PointLocation types (Obb, Frustum) are not Copy as well, and the same applies to them. I think it's nice to encourage passing by reference since for an Aabb3 it's probably more efficient (48 bytes).